### DISCUSSION TITLE
Support multiple GHC versions indexed by compiler version

### DISCUSSION THREAD

**Discussion by paulvictor** [srid/haskell-flake#104] [Ideas]
I would like to make a proposal to add the flake output to be parameterized on the ghc version like in  
 https://github.com/paulvictor/try-ghcjs/blob/main/flake.nix#L51-L160

Is something like this possible?

**Comment by srid** [comment]
You can define multiple projects each corresponding to a GHC version. See for example,

https://github.com/srid/ema/blob/ddf0a92d9de8e06906dc5ada2dfc9a00a3153a7e/flake.nix

Separating on a project-basis is better because when changing GHC version, usually you end up changing the overrides as well (as shown by the Ema example), sometimes even the devShell tools.

The common parts can be defined as [a module of its own and imported](https://haskell.flake.page/modules) under the two GHC version projects to respect the DRY principle.

**Reply by srid** [reply]
Note to self: this would be a good candidate for documentation.

**Reply by paulvictor** [reply]
Thanks, this makes sense. Will try it for now
